### ISSUE TITLE
HyperParameterOptimizer fails because bask task hyper parameters types are not copied over

### ISSUE THREAD

**Issue by reiffd7** [clearml/clearml#975]
## Describe the bug
When I run HyperParameterOptimizer from clearml.automation, each child "experiment" task fails because the hyper parameter types are not copied over correctly. Each type for the hyper parameters is converted to strings.

base task:
<sm-section-header _ngcontent-ecp-c361="" _nghost-ecp-c260="" style="padding: 12px 0px; margin: 0px; outline: none !important; box-sizing: border-box; display: flex; justify-content: space-between; height: 35px; color: rgb(90, 101, 142); font-family: Heebo, sans-serif, sans-serif; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><h4 _ngcontent-ecp-c260="" style="padding: 0px; margin: 0px; outline: none !important; box-sizing: border-box; font-size: 14px; font-weight: 700; display: block; position: relative; color: rgb(90, 101, 142);">GENERAL</h4></sm-section-header><sm-experiment-execution-parameters _ngcontent-ecp-c361="" class="form-section" _nghost-ecp-c360="" style="padding: 0px; margin: 0px; outline: none !important; box-sizing: border-box; display: block; height: calc(100% - 42px); color: rgb(90, 101, 142); font-family: Heebo, sans-serif, sans-serif; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><form _ngcontent-ecp-c360="" novalidate="" class="ng-untouched ng-pristine ng-valid" style="padding: 0px; margin: 0px; outline: none !important; box-sizing: border-box;"><cdk-virtual-scroll-viewport _ngcontent-ecp-c360="" itemsize="60" minbufferpx="280" maxbufferpx="560" class="cdk-virtual-scroll-viewport form-container cdk-virtual-scrollable cdk-virtual-scroll-orientation-vertical" style="padding: 0px; margin: 0px; outline: none !important; display: block; position: relative; transform: translateZ(0px); box-sizing: border-box; overflow: auto; will-change: scroll-position; contain: strict; height: 0px;"><div class="cdk-virtual-scroll-content-wrapper" style="padding: 0px; margin: 0px; outline: none !important; box-sizing: border-box; position: absolute; top: 0px; left: 0px; contain: content; min-width: 100%; transform: translateY(0px);"></div><div class="cdk-virtual-scroll-spacer" style="padding: 0px; margin: 0px; outline: none !important; box-sizing: border-box; height: 1px; transform-origin: 0px 0px; flex: 0 0 auto;"></div></cdk-virtual-scroll-viewport></form><div _ngcontent-ecp-c360="" class="table-container ng-star-inserted" style="padding: 0px 0px 6px; margin: 0px; outline: none !important; box-sizing: border-box; height: 617.127px;"><sm-table _ngcontent-ecp-c360="" columnresizemode="fit" _nghost-ecp-c146="" style="padding: 0px; margin: 0px; outline: none !important; box-sizing: border-box;"><p-table _ngcontent-ecp-c146="" datakey="id" class="p-element table ng-star-inserted" style="padding: 0px; margin: 0px; outline: none !important; box-sizing: border-box; color: rgb(220, 224, 238);"><div id="pr_id_3" class="p-datatable p-component p-datatable-hoverable-rows p-datatable-scrollable p-datatable-scrollable-vertical p-datatable-flex-scrollable p-datatable-responsive-stack" style="padding: 0px; margin: 0px; outline: none !important; box-sizing: border-box; position: relative; display: flex; flex-direction: column; height: 611.128px; flex: 1 1 0%;"><div class="p-datatable-wrapper" style="padding: 0px; margin: 0px; outline: none !important; box-sizing: border-box; position: relative; overflow: auto; display: flex; flex-direction: column; flex: 1 1 0%; height: 611.128px;"><p-scroller class="p-scroller-viewport p-element ng-star-inserted" style="padding: 0px; margin: 0px; outline: none 0px; flex: 1 1 0%; box-sizing: border-box;"><div tabindex="0" class="p-scroller ng-star-inserted" style="padding: 0px; margin: 0px; outline: none 0px; box-sizing: border-box; position: relative; overflow: auto; contain: strict; transform: translateZ(0px); will-change: scroll-position; height: 611.128px;">

batch_size | 100 //type Int
-- | --

optimization task:
<sm-section-header _ngcontent-ecp-c361="" _nghost-ecp-c260="" style="padding: 12px 0px; margin: 0px; outline: none !important; box-sizing: border-box; display: flex; justify-content: space-between; height: 35px; color: rgb(90, 101, 142); font-family: Heebo, sans-serif, sans-serif; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><h4 _ngcontent-ecp-c260="" style="padding: 0px; margin: 0px; outline: none !important; box-sizing: border-box; font-size: 14px; font-weight: 700; display: block; position: relative; color: rgb(90, 101, 142);">GENERAL</h4></sm-section-header><sm-experiment-execution-parameters _ngcontent-ecp-c361="" class="form-section" _nghost-ecp-c360="" style="padding: 0px; margin: 0px; outline: none !important; box-sizing: border-box; display: block; height: calc(100% - 42px); color: rgb(90, 101, 142); font-family: Heebo, sans-serif, sans-serif; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><form _ngcontent-ecp-c360="" novalidate="" class="ng-untouched ng-pristine ng-valid" style="padding: 0px; margin: 0px; outline: none !important; box-sizing: border-box;"><cdk-virtual-scroll-viewport _ngcontent-ecp-c360="" itemsize="60" minbufferpx="280" maxbufferpx="560" class="cdk-virtual-scroll-viewport form-container cdk-virtual-scrollable cdk-virtual-scroll-orientation-vertical" style="padding: 0px; margin: 0px; outline: none !important; display: block; position: relative; transform: translateZ(0px); box-sizing: border-box; overflow: auto; will-change: scroll-position; contain: strict; height: 0px;"><div class="cdk-virtual-scroll-content-wrapper" style="padding: 0px; margin: 0px; outline: none !important; box-sizing: border-box; position: absolute; top: 0px; left: 0px; contain: content; min-width: 100%; transform: translateY(0px);"></div><div class="cdk-virtual-scroll-spacer" style="padding: 0px; margin: 0px; outline: none !important; box-sizing: border-box; height: 1px; transform-origin: 0px 0px; flex: 0 0 auto;"></div></cdk-virtual-scroll-viewport></form><div _ngcontent-ecp-c360="" class="table-container ng-star-inserted" style="padding: 0px 0px 6px; margin: 0px; outline: none !important; box-sizing: border-box; height: 617.127px;"><sm-table _ngcontent-ecp-c360="" columnresizemode="fit" _nghost-ecp-c146="" style="padding: 0px; margin: 0px; outline: none !important; box-sizing: border-box;"><p-table _ngcontent-ecp-c146="" datakey="id" class="p-element table ng-star-inserted" style="padding: 0px; margin: 0px; outline: none !important; box-sizing: border-box; color: rgb(220, 224, 238);"><div id="pr_id_3" class="p-datatable p-component p-datatable-hoverable-rows p-datatable-scrollable p-datatable-scrollable-vertical p-datatable-flex-scrollable p-datatable-responsive-stack" style="padding: 0px; margin: 0px; outline: none !important; box-sizing: border-box; position: relative; display: flex; flex-direction: column; height: 611.128px; flex: 1 1 0%;"><div class="p-datatable-wrapper" style="padding: 0px; margin: 0px; outline: none !important; box-sizing: border-box; position: relative; overflow: auto; display: flex; flex-direction: column; flex: 1 1 0%; height: 611.128px;"><p-scroller class="p-scroller-viewport p-element ng-star-inserted" style="padding: 0px; margin: 0px; outline: none 0px; flex: 1 1 0%; box-sizing: border-box;"><div tabindex="0" class="p-scroller ng-star-inserted" style="padding: 0px; margin: 0px; outline: none 0px; box-sizing: border-box; position: relative; overflow: auto; contain: strict; transform: translateZ(0px); will-change: scroll-position; height: 611.128px;">

batch_size | 100 //type String
-- | --

</div></p-scroller></div></div></p-table></sm-table></div></sm-experiment-execution-parameters>

## To reproduce
Just followed the tutorial @ https://github.com/allegroai/clearml/blob/master/examples/optimization/hyper-parameter-optimization/hyper_parameter_optimizer.py

but replaced the base task id with one of my tasks.

## Expected behaviour
I would expect the correct hyper parameter types to copy over from the base task to the hyper parameter search child task. I know that task.get_parameters() has a cast parameter that can be used to retrieve the correct types, but this does not seem to be an option when conducting a hyper parameter search. I was able to fix the issue locally by editing clearml/automation/job.py.
In line 410:
`        hyper_params = task.get_parameters(cast=True) if params_override is None else params_override
`
In line 541:
`            task_params = base_temp_task.get_parameters(cast=True, backwards_compatibility=False)
`

But, I would like to be able to pass cast=True to the HyperParameterOptimizer method.

## Environment
* Server type (self hosted \ app.clear.ml): app.clear.ml
* ClearML SDK Version: 1.10.3
* Python Version: 3.7.16
* OS (Windows \ Linux \ Macos): Linux

**Comment by phineasng** [comment]
Any update on this, or did you manage to find a less invasive workaround by any chance @reiffd7 ?

**Comment by AlexandruBurlacu** [comment]
Hello @reiffd7 and @phineasng, sorry for taking such long time. I believe we solved this issue (wrong casting of configuration values) in more recent ClearML versions. Try to install the latest version, which is 1.11.0, and please let us know whether it solves the bug

**Comment by wxdrizzle** [comment]
Hi @AlexandruBurlacu , seems that in recent version this problem still exists. I created a new issue #1238 for this several days ago and today just found this issue is exactly what I have.

**Comment by eugen-ajechiloae-clearml** [comment]
Hi @wxdrizzle ! We have prepared a fix for this issue and we will soon release it.

**Comment by wxdrizzle** [comment]
> Hi @wxdrizzle @reiffd7 ! We have prepared a fix for this issue and we will soon release it.

Happy to know this. Thank you! I'll update once I can try the new release.

**Comment by eugen-ajechiloae-clearml** [comment]
Hi @reiffd7 @phineasng @wxdrizzle ! We have released an RC (and we will soon release an official version) that addresses this issue. Can you please try it out? To install it, do `pip install clearml==1.16.0rc0`

**Comment by wxdrizzle** [comment]
> Hi @reiffd7 @phineasng @wxdrizzle ! We have released an RC (and we will soon release an official version) that addresses this issue. Can you please try it out? To install it, do `pip install clearml==1.16.0rc0`

Hi @eugen-ajechiloae-clearml , I tried this and confirm that in my case it solved the problem perfectly. I'll close my issue. Thank you so much!!!

**Comment by pollfly** [comment]
Hey @reiffd7! Just letting you know that this issue has been resolved in the recently released [v1.16.0](https://github.com/allegroai/clearml/releases/tag/v1.16.0). Let us know if there are any issues :)
